### DISCUSSION TITLE
Universal2 arch build on Mac Intel

### DISCUSSION THREAD

**Discussion by Alias419** [pyinstaller/pyinstaller#7672] [Help]
When specifying arch universal2 on build with icon file specified, currently receiving error below:

PIL/_imagingft.cpython-311-darwin.so is not a fat binary!

What would be a workaround for this, pillow is only needed on the build with pyinstaller.

**Comment by rokm** [comment]
> When specifying arch universal2 on build with icon file specified, currently receiving error below:

I doubt this has anything to do with the icon - your application (or one of its dependencies) is using `pillow`, hence it is being collected.

You can only build a `universal2` application if your python is `universal2` AND all non-pure-python packages (i.e., the ones with binary extensions) are `universal2`. In this case, `pillow` provides separate thin-binary wheels for x86_64 and arm64, so that won't work. 

The work-around would be to either rebuild `pillow` as `universal2` package yourself. Or to manually download both arch wheels, unpack them, stitch the thin-binaries together into fat binaries using `lipo` (for all extensions in `pillow`), and then replace the thin binaries in your environment with the manually-produced fat ones.
